<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
    "http://www.w3.org/TR/html4/strict.dtd">
<html lang="ja">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=Shift_JIS">
    <title>Maya : Documentation : appendix D. JavaScript を書くときの Tips</title>
    <link rel="STYLESHEET" href="./maya.css" type="text/css">
</head>
<body>
<div class="header">
<div>Maya : JavaServer Templates</div>
</div>

<div class="navi">
<a href="./index.html">↑index</a>
</div>

<div style="font-size: xx-large; background-color: red; color: white">作成中</div>

<div class="main">

<h1>appendix D. JavaScript を書くときの Tips</h1>


<h2>Java のクラスを使う</h2>

<p>Java のクラスは、<code>java.lang.String</code> や <code>java.util.ArrayList</code> のようにパッケージが "<code>java.</code>" で始まるクラスであれば、完全修飾名をそのまま書くだけで利用できます。しかしそれ以外のパッケージ ("<code>org.</code>" や "<code>com.</code>" など) の場合、先頭に "<code>Packages.</code>" を付ける必要があります。</p>

<pre>
例:
var list = new java.util.ArrayList();
var myClass = new <strong>Packages.</strong>org.example.MyClass();
</pre>


<h2>例外を処理する</h2>

<p>JavaScript にも try 〜 catch 構文はありますので、Java で発生した例外を JavaScript で処理することもできます。Java でアプリケーションを作るときには Java 内で処理するものでしょうが、デバッグ時には役立つかもしれません。</p>

<pre>
var catched = "";
<strong>try {</strong>
    var list = myLogic.findByName(name);
<strong>} catch (e) {</strong>
    catched = e;
<strong>}</strong>

&lt;m:write id="error" value="&#36;{ catched }" /&gt;
</pre>


</div>

<div class="footer">
&copy;2004-2005, The Seasar Foundation and the others, all rights reserved.
</div>

</body>
</html>
